### PR TITLE
Provide the V3 API of Ibis server for the wren-core rust beta version

### PR DESCRIPTION
**Pull request by goldmedal**
# Description
- V3 APIs only include `query`, `dry-run`, `dry-plan` and `validate` API
- All of the API endpoints are the same as `v2` except the prefix is `v3`.
- The `validate` API has some issues that need to be fixed. Disable the test first.
```
# TODO: enable this test after supporting the default catalog and default schema
```
- Only test Postgres currently.

### CODE REVIEW

File: ibis-server/app/routers/v3/connector.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        )
+        connector = Connector(data_source, dto.connection_info, dto.manifest_str)
+        if dry_run:
+            connector.dry_run(rewritten_sql)
+            return Response(status_code=204)
+        return JSONResponse(
+            to_json(connector.query(rewritten_sql, limit=limit), dto.column_dtypes)
+        )
+    except Exception as e:
+        return Response(str(e), status_code=422)
```
[the remark this answers is not in the document]
Comment (goldmedal) [reply ibis-server/app/routers/v3/connector.py:42]: In this case, we only return the error message when the query fails.

File: ibis-server/app/routers/v3/connector.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+
+@router.post("/dry-plan")
+@log_dto
+def dry_plan(dto: DryPlanDTO) -> str:
+    try:
+        return EmbeddedEngineRewriter(dto.manifest_str).rewrite(dto.sql)
+    except Exception as e:
+        return Response(str(e), status_code=422)
```
[the remark this answers is not in the document]
Comment (goldmedal) [reply ibis-server/app/routers/v3/connector.py:51]: Same as above

File: ibis-server/app/routers/v3/connector.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+
+@router.post("/{data_source}/dry-plan")
+@log_dto
+def dry_plan_for_data_source(data_source: DataSource, dto: DryPlanDTO) -> str:
+    try:
+        return EmbeddedEngineRewriter(dto.manifest_str, data_source).rewrite(dto.sql)
+    except Exception as e:
+        return Response(str(e), status_code=422)
```
[the remark this answers is not in the document]
Comment (goldmedal) [reply ibis-server/app/routers/v3/connector.py:60]: Same as above

File: ibis-server/app/routers/v3/connector.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+def validate(data_source: DataSource, rule_name: str, dto: ValidateDTO) -> Response:
+    try:
+        validator = Validator(
+            Connector(data_source, dto.connection_info, dto.manifest_str),
+            EmbeddedEngineRewriter(dto.manifest_str, data_source),
+        )
+        validator.validate(rule_name, dto.parameters)
+        return Response(status_code=204)
+    except Exception as e:
+        return Response(str(e), status_code=422)
```
[the remark this answers is not in the document]
Comment (goldmedal) [reply ibis-server/app/routers/v3/connector.py:74]: Same as above

**Review by grieve54706 — APPROVED** [review]

**Pull request merged**